src/test/java/seedu/address/model/VersionedAppointmentBookTest.java

```diff
@@ -0,0 +1,314 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalAppointments.ALICE_APPOINTMENT;
+import static seedu.address.testutil.TypicalAppointments.BENSON_APPOINTMENT;
+import static seedu.address.testutil.TypicalAppointments.HOON_APPOINTMENT;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.exceptions.NoRedoableStateException;
+import seedu.address.model.exceptions.NoUndoableStateException;
+import seedu.address.testutil.AppointmentBookBuilder;
+
+public class VersionedAppointmentBookTest {
+
+    private final ReadOnlyAppointmentBook appointmentBookWithAmy = new AppointmentBookBuilder()
```

Comment:
Would it be better if the name of the appointmentBook is consistent with the patient's name of the appointment? I was a little bit confused when I saw `appointmentBookWithAmy` has appointment `ALICE_APPOINTMENT` added. Perhaps I misunderstood this part. Let me know what you think 👍